os/kernel/init/os_start.c

```diff
@@ -447,6 +650,13 @@ void os_start(void)
 	}
 #endif
 
+	/* Disables context switching because we need take the memory manager
+	 * semaphore on this CPU so that it will not be available on the other
+	 * CPUs until we have finished initialization.
+	 */
+
+	sched_lock();
```

Comment:
Do you call `sched_lock` instead of `enter_critical_section` here?

Reply:
It is in accordance with nuttx smp code.